libraries/Network/src/NetworkUdp.cpp

```diff
@@ -292,7 +292,10 @@ void NetworkUDP::flush() {}
 
 int NetworkUDP::parsePacket() {
   if (rx_buffer) {
-    return 0;
+    if (rx_buffer->full()) { // packet was not read yet
```

Comment:
See both implementations of `room()` and `full()`:
```c++
size_t cbuf::room() const {
  size_t _room = 0;
  if (_buf != NULL) {
    _room = xRingbufferGetCurFreeSize(_buf);
  }
  return _room;
}

bool cbuf::full() const {
  return room() == 0;
}
```

As you can see, when checking for `full()`, you only check if there is no more room left.
However I do not know how these buffers are allocated.
For example if I process a packet of 7 bytes, what guarantee do we have the buffer is full then and there isn't any extra byte of free space?

See also the note in the comments in the header file where the buffer is created:
```c++
/**
 * @brief       Create a ring buffer
 *
 * @param[in]   xBufferSize Size of the buffer in bytes. Note that items require
 *              space for a header in no-split/allow-split buffers
 * @param[in]   xBufferType Type of ring buffer, see documentation.
 *
 * @note    xBufferSize of no-split/allow-split buffers will be rounded up to the nearest 32-bit aligned size.
 *
 * @return  A handle to the created ring buffer, or NULL in case of error.
 */
RingbufHandle_t xRingbufferCreate(size_t xBufferSize, RingbufferType_t xBufferType);
```

Reply:
There is however another obvious bug in the NetworkUDP.cpp which may need some fixing...

```c++
  if (len > 0) {
    rx_buffer = new (std::nothrow) cbuf(len);
    rx_buffer->write(buf, len);
  }
```

Reply:
I tested the cbuf class to check if it behaves as I expected.

Reply:
But the note at the `xRingbufferCreate` function clearly states it will round up `xBufferSize` to the nearest 4-byte.
If this happens, then you can't use the `full()` function to detect whether it is still untouched.

Reply:
RINGBUF_TYPE_BYTEBUF is used here. the note is for RINGBUF_TYPE_NOSPLIT and RINGBUF_TYPE_ALLOWSPLIT

Reply:
Ah check!